### ISSUE TITLE
Add more pandas / row-based  / tabular mapping examples to Datasets docs

### ISSUE THREAD

**Issue by worldveil** [ray-project/ray#22084]
Datasets treats everything is very map based, as opposed to index or iteration based like pandas or others.

If we don't point users that direction they're gonna have a confusing time, and wonder if Ray is even a good tool for their usecase if they are doing dataframe transformations on feature data for ML.

From @clarkzinzow : "Add in ctrl-f keywords like "row" and "column", more (tabular) examples, etc."

**Comment by worldveil** [comment]
Just wanted to check in here, we're still targeting end of March for this?

**Comment by clarkzinzow** [comment]
@worldveil Yep, it's on the list! https://github.com/ray-project/ray/issues/18262

**Comment by clarkzinzow** [comment]
This was done in the Datasets GA docs push.
